### PR TITLE
Feat/identifier

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/model/representation/io/detail/geode_brep_input.h [hunk 1]
```diff
@@ -48,6 +49,11 @@ namespace geode
         void load_brep_files( absl::string_view directory )
         {
             BRepBuilder builder{ brep() };
+            builder.load_identifier( directory );
+            if( brep().name() == Identifier::DEFAULT_NAME )
```
Comment (BotellaA) [inline include/geode/model/representation/io/detail/geode_brep_input.h:53]: Should we do the same for Mesh inputs?

File: include/geode/basic/identifier.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        void set_name( absl::string_view name, IdentifierKey );
+
+        void load_identifier( absl::string_view directory, IdentifierKey );
+
+    protected:
+        Identifier();
+
+        void set_id( uuid id );
+
+        void set_name( absl::string_view name );
```
Comment (panquez) [inline include/geode/basic/identifier.h:66]: useful?

File: include/geode/basic/identifier.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace geode
+{
+    class opengeode_basic_api Identifier
+    {
+        PASSKEY( IdentifierBuilder, IdentifierKey );
+        friend class bitsery::Access;
+
+    public:
+        static constexpr auto DEFAULT_NAME = "unknown";
```
Comment (panquez) [inline include/geode/basic/identifier.h:44]: Is "unknown" a good choice? Proposal: "default_name", "no_name", "unnamed", ...

**Review by panquez — APPROVED** [review]

File: include/geode/basic/identifier.h (hunk 2 above)
Comment (BotellaA) [reply include/geode/basic/identifier.h:66]: yes, it is used in bitsery compatibility. But I could also use a Builder inside Bitsery if you prefer.

**Pull request merged**